pkg/services/apiserver/config.go

```diff
@@ -5,15 +5,20 @@ import (
 	"net"
 	"path/filepath"
 	"strconv"
+	"strings"
 
 	"github.com/grafana/grafana/pkg/services/apiserver/options"
 	"github.com/grafana/grafana/pkg/services/featuremgmt"
 	"github.com/grafana/grafana/pkg/setting"
 )
 
-func applyGrafanaConfig(cfg *setting.Cfg, features featuremgmt.FeatureToggles, o *options.Options) {
+func applyGrafanaConfig(cfg *setting.Cfg, features featuremgmt.FeatureToggles, o *options.Options) error {
 	defaultLogLevel := 0
-	ip := net.ParseIP(cfg.HTTPAddr)
+	ipStr := strings.TrimSuffix(strings.TrimPrefix(cfg.HTTPAddr, "["), "]")
```

Comment:
Not sure that we actually need this, since right now ipv6 addresses enclosed in square brackets don't work (afaik) and we're better off being strict about input and requiring a valid ip address

Reply:
i see, but some people use IP in [] brackets. it will return some error.